### ISSUE TITLE
Some flatpak applications need more than 90 seconds to start and hog a cpu core when "isolate workspaces" is enabled

### ISSUE THREAD

**Issue by crazy-weasel** [home-sweet-gnome/dash-to-panel#2192]
**Describe the bug**
With the "Isolate Workspaces" feature enabled, some flatpak apps need >90 seconds to start. It does not matter if the application is started from the panel, or from the applications menu. If the flatpak app is started via the terminal, it starts immediately.

**Linux distribution and version**
Debian Testing / Trixie

**GNOME Shell version**
GNOME Shell 46.4

**Dash-to-Panel version**
63

**Where was Dash-to-Panel installed from?**
The GNOME extensions website

**Screenshots / Video captures**
https://github.com/user-attachments/assets/923e6a94-adcd-405e-af48-22dc685394da

In the video:
Starting chromium at the 4 seconds mark, becomes visible at 01:43
Then disabling the feature and starting Chromium again at 01:55 and is visibly shortly after

**Additional Info**
Not all flatpak apps are affected. Reproducible on my machine with Chromium and Inkscape. Thunderbird starts normally, but the "Write New message" action from the context menu is also having the same issue.

**Comment by archekb** [comment]
Ubuntu 24.10 the same problem.
Ubuntu 24.04 have no problem.

**Comment by charlesg99** [comment]
I don't think this is related to dash to panel. Starting apps is done by gnome shell, and the delay before the chromium window appears isn't affected by dash to panel in any way. I don't use snaps or flatpaks, but I would guess this delay is caused by the fact that the application hasn't been started yet. In your test, try a third time after re enabling workspace isolation.

**Comment by archekb** [comment]
Hi, @charlesg99 

Unfortunately, this is bug of Dash to Panel, because, the same function `Isolate Workspaces` at Ubuntu Dock works fine. Yesterday I tried debug it, by this script:
```
#!/bin/sh -e

#export G_MESSAGES_DEBUG=all
export MUTTER_DEBUG_DUMMY_MODE_SPECS=1366x768
export SHELL_DEBUG=all

dbus-run-session -- gnome-shell --nested --wayland
```
and only one difference between on and off `Isolate Workspaces` is 
 
> F: Can't get document portal: GDBus.Error:org.freedesktop.DBus.Error.NoReply: Message recipient disconnected from message bus without replying

 I didn't found any solution and yesterday reinstall Ubuntu back to 24.04 and now Dash to Panel with `Isolate Workspaces` works ok for me.

**Comment by archekb** [comment]
@charlesg99 One more, maybe you right, if you unpin the App from the panel and run it from gnome app screen, result will the same, about 90 seconds, but if we turn off `Isolate Workspaces` at Dash To Panel it will start immediately. With Ubuntu dock apps runs immediately, not matter is `Isolate Workspaces` turned on or off.

**Comment by charlesg99** [comment]
Thanks, I can reproduce it on Ubuntu 24.10. Unfortunately this is a gnome-shell bug and it also happens with the ubuntu-dock. I've opened an issue on the gnome-shell repository, hopefully they'll fix it: 

https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/8188

**Comment by archekb** [comment]
@charlesg99 Thank you
